### ISSUE TITLE
Wazuh Agent resource utilization on rocky 8

### ISSUE THREAD

**Issue by shegzon** [wazuh/wazuh#19874]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.5.3 | Wazuh Agent | Manager/Agent | Packages/Sources | rocky 8 |

Hello Team,

There is an issue discovered on Wazuh agent, even when the service is stopped, wazuh-syscheckd starts by itself and eats lots of CPU. Blocks wordpress from installing plugins.

Kindly find below the output of "top" for the CPU utilization and also after enabling debugging on the affected endpoint.

1. Output of **top** command:
![syscheckd](https://github.com/wazuh/wazuh/assets/15922108/b47aebbf-5366-44f1-b13c-51335ff67b7e)

2. Status of service after being stopped
![syscheckd2](https://github.com/wazuh/wazuh/assets/15922108/7ff7e1b3-986c-4529-9870-cf2c9eb35363)

3. Output after enabling debugging:
![debugging1](https://github.com/wazuh/wazuh/assets/15922108/c5306cdd-08e5-43e9-800f-16cea05efae3)
![debugging2](https://github.com/wazuh/wazuh/assets/15922108/780cc3e8-3076-4f4d-a8a8-8299ab04d103)
![debugging3](https://github.com/wazuh/wazuh/assets/15922108/0375a5b5-d561-4f2e-829d-173ccac8a9fa)
![debugging4](https://github.com/wazuh/wazuh/assets/15922108/02f80613-d977-49e7-a516-be19eac8785e)

**Comment by vikman90** [comment]
@shegzon FIM needs to build up its file inventory on the agent's startup, that's why it consumes lots of resources. However, the FIM process (_wazuh-syscheckd_) runs with less priority (30), yielding the system (CPU / disk) to other processes that require it.

```xml
<syscheck>
  <process_priority>10</process_priority>
</syscheck>
```

This adds +10 to the process priority: 20 [default] + 10 = 30. The higher the value, the lower the priority.

In addition, you can slow down the file analysis by limiting the number of files analyzed per second:

```xml
<syscheck>
  <max_files_per_second>10</max_files_per_second>
</syscheck>
```

On the other hand, this log:

> Couldn't insert '...' entry into DB. The DB is full.

You've reached the 100K file limit. You can extend it with:

```xml
<file_limit>
  <enabled>yes</enabled>
  <entries>200000</entries>
</file_limit>
```

Please refer to https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/syscheck.html

However, these logs make me think that the FIM settings are very aggressive. I'm curious, can you share your settings?

Anyway, in summary:

- We know this issue.
- We can use the settings above to reduce the process footprint in the system.

Best!

**Comment by shegzon** [comment]
Hello @vikman90,

Thanks a lot for the detailed response. I earlier requested for the configuration file from the user that made the complain so that I could try and replicate in the lab but no response yet. Once he reverts, I'll update you accordingly. Thanks again.

Best regards.
